syntax = "proto3";

message Person{
    message Contact{
        string mail = 1;
        string phone = 2;
    }
    enum PersonType{
        PERSON_TYPE_UNSET = 0;
        PERSON_TYPE_LEGEND = 1;
        PERSON_TYPE_ULTRALEGEND = 2;
    }
    int64 id = 1;
    string name = 2;
    int32 age = 3;
    Contact contact = 4;
    PersonType person_type = 5;
}

message Employee{
    int32 emp_id = 1;
    string emp_name = 2;
    Person.Contact emp_contact = 3;
}

